### PR TITLE
Fix testGenerateAndSignMetadata in FIPS mode

### PR DESCRIPTION
**Pull request by jkakavas**
BC provider throws different error message on signature
validation failure

### CODE REVIEW

File: x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/idp/SamlMetadataGeneratorTests.java [hunk 1]
```diff
@@ -101,6 +101,7 @@ public void testGenerateMetadata() throws Exception {
         assertValidXml(xml);
     }
 
+
```
Comment (tvernum) [inline x-pack/plugin/identity-provider/src/test/java/org/elasticsearch/xpack/idp/saml/idp/SamlMetadataGeneratorTests.java]: Nit: extra line

**Review by tvernum — APPROVED** [review]
LGTM

**Comment by jkakavas** [comment]
@elasticmachine update branch

**Comment by jkakavas** [comment]
@elasticmachine test this please

**Comment by jkakavas** [comment]
probably https://github.com/elastic/elasticsearch/pull/54321 will fix this

**Comment by jkakavas** [comment]
@elasticmachine update branch

**Pull request merged**